bottle.py

```diff
@@ -2773,6 +2781,11 @@ def static_file(filename, root,
         check or continue partial downloads) are also handled automatically.
     """
 
+    if isinstance(root, Path):
+        root = str(root)
```

Comment:
You can skip the type check and just call `str(var)` or `Path(var)` to get what you need. This may even be cheaper than the type check :)